/* Copyright 2020-2023 The Loimos Project Developers.
 * See the top-level LICENSE file for details.
 *
 * SPDX-License-Identifier: MIT
 */

syntax = "proto3";
package loimos.proto;

message DataField {
  string field_name = 1;

  message Empty {
  }

  message Category {
    repeated int32 values = 1;
  }

  oneof data_type {
    Empty unique_id = 2;
    Empty foreign_id = 3;
    Empty start_time = 4;
    Empty duration = 5;
    Empty bool = 6;
    Empty b10int = 7;
    Empty int32 = 8;
    Empty double = 9;
    Empty b10double = 10;
    Empty ignore = 11;
    Empty label = 12;
    Category category = 13;
  }
}

message CSVDefinition {
  repeated DataField fields = 1;
}
